web/lib/src/helpers/cross_origin.dart

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:js_interop';
+
+import '../dom.dart' show HTMLIFrameElement, Location, Window;
+
+// Includes all the allowed and necessary APIs from the W3 spec located here:
```

Comment:
(optional) consider turning it into a dart doc to the extension type declaration below?